web/ui_utils.js

```diff
@@ -318,44 +327,226 @@ function getVisibleElements(scrollEl, views, sortByVisibility = false) {
+}
+
+/**
+ * Same as above, only with top/bottom/height/y and left/right/width/x
+ * reversed. This fact is enforced with an automated test in ui_utils_spec.js.
+ */
+function getVisibleElementsHoriz(scrollEl, views, sortByVisibility = false) {
+  let left = scrollEl.scrollLeft, right = left + scrollEl.clientWidth;
+  let top = scrollEl.scrollTop, bottom = top + scrollEl.clientHeight;
+
+  function isElementRightAfterViewLeft(view) {
+    let element = view.div;
+    let elementRight =
+      element.offsetLeft + element.clientLeft + element.clientWidth;
+    return elementRight > left;
+  }
+
+  let visible = [], view, element;
+  let currentWidth, viewWidth, hiddenWidth;
+  let currentHeight, viewHeight, hiddenHeight;
+  let percentVisible;
+  let firstVisibleElementInd = views.length === 0 ? 0 :
+    binarySearchFirstItem(views, isElementRightAfterViewLeft);
+
+  if (views.length > 0) {
+    // See comments in getVisibleElements.
+    let backtrack = false, pageLeft, elt;
+    elt = views[firstVisibleElementInd].div;
+    pageLeft = elt.offsetLeft + elt.clientLeft;
+    if (pageLeft < left) {
+      backtrack = true;
+    } else if (firstVisibleElementInd >= 2) {
+      elt = views[firstVisibleElementInd - 1].div;
+      pageLeft = elt.offsetLeft + elt.clientLeft;
+
+      elt = views[firstVisibleElementInd - 2].div;
+      if (elt.offsetLeft + elt.clientLeft + elt.clientWidth > pageLeft) {
+        firstVisibleElementInd -= 2;
+        backtrack = true;
+      }
+    }
+    if (backtrack) {
+      for (; firstVisibleElementInd > 0; --firstVisibleElementInd) {
+        elt = views[firstVisibleElementInd - 1].div;
+        if (elt.offsetLeft + elt.clientLeft + elt.clientWidth <= pageLeft) {
+          break;
+        }
+      }
+    }
+  }
+
+  let lastRight = -1, viewBottom, viewRight;
+
+  for (let i = firstVisibleElementInd, ii = views.length; i < ii; i++) {
+    view = views[i];
+    element = view.div;
+    currentWidth = element.offsetLeft + element.clientLeft;
+    viewWidth = element.clientWidth;
+    viewRight = currentWidth + viewWidth;
+
+    if (viewRight <= left) {
+      continue;
+    }
+    if (lastRight === -1) {
+      if (viewRight >= right) {
+        lastRight = viewRight;
+      }
+    } else if (currentWidth > lastRight) {
+      break;
+    }
+    if (currentWidth >= right) {
+      continue;
+    }
+
+    currentHeight = element.offsetTop + element.clientTop;
+    viewHeight = element.clientHeight;
+    viewBottom = currentHeight + viewHeight;
+    if (viewBottom < top || currentHeight > bottom) {
+      continue;
+    }
+    hiddenWidth = Math.max(0, left - currentWidth) +
+      Math.max(0, viewRight - right);
+    hiddenHeight = Math.max(0, top - currentHeight) +
+      Math.max(0, viewBottom - bottom);
+    percentVisible = ((viewWidth - hiddenWidth) * (viewHeight - hiddenHeight) *
+      100 / viewWidth / viewHeight) | 0;
+
+    visible.push({
+      id: view.id,
+      y: currentHeight,
+      x: currentWidth,
```

Comment:
You've answered your own question. 😄 A dumb search-and-replace swap of `currentWidth` and `currentHeight` has to be accompanied by a search-and-replace swap of `x:` and `y:`, which ends up effectively swapping the lines. (In fact, [this test](/mozilla/pdf.js/pull/9208/files/63f6328ad87bb7ffc548abd335ba962b67a6ae5a#diff-c7f5d75f7b6b0f13392f9d4c49a0c990R485) performs that search-and-replace, and would fail if the lines were put back in `x`, `y` order.)